test/hpke_test.c

```diff
@@ -0,0 +1,1545 @@
+    0xd8, 0xf5, 0xbe, 0xd0, 0xe8, 0x6e, 0xb8, 0x84,
+    0x1c, 0xe4, 0x89, 0x2e, 0x0f, 0xc3, 0x87, 0xbb,
+    0xdb, 0xfe, 0x16, 0x0d, 0x58, 0x9c, 0x89, 0x2d,
+    0xd4, 0xb1, 0x46, 0x4a, 0xc3, 0x51, 0xc5, 0x6f,
+    0xb6
+};
+
+/* from RFC9180 Appendix A.6.1 */
+static unsigned char ikmp521[] = {
+    0x7f, 0x06, 0xab, 0x82, 0x15, 0x10, 0x5f, 0xc4,
+    0x6a, 0xce, 0xeb, 0x2e, 0x3d, 0xc5, 0x02, 0x8b,
+    0x44, 0x36, 0x4f, 0x96, 0x04, 0x26, 0xeb, 0x0d,
+    0x8e, 0x40, 0x26, 0xc2, 0xf8, 0xb5, 0xd7, 0xe7,
+    0xa9, 0x86, 0x68, 0x8f, 0x15, 0x91, 0xab, 0xf5,
+    0xab, 0x75, 0x3c, 0x35, 0x7a, 0x5d, 0x6f, 0x04,
+    0x40, 0x41, 0x4b, 0x4e, 0xd4, 0xed, 0xe7, 0x13,
+    0x17, 0x77, 0x2a, 0xc9, 0x8d, 0x92, 0x39, 0xf7,
+    0x09, 0x04
+};
+static unsigned char pubp521[] = {
+    0x04, 0x01, 0x38, 0xb3, 0x85, 0xca, 0x16, 0xbb,
+    0x0d, 0x5f, 0xa0, 0xc0, 0x66, 0x5f, 0xbb, 0xd7,
+    0xe6, 0x9e, 0x3e, 0xe2, 0x9f, 0x63, 0x99, 0x1d,
+    0x3e, 0x9b, 0x5f, 0xa7, 0x40, 0xaa, 0xb8, 0x90,
+    0x0a, 0xae, 0xed, 0x46, 0xed, 0x73, 0xa4, 0x90,
+    0x55, 0x75, 0x84, 0x25, 0xa0, 0xce, 0x36, 0x50,
+    0x7c, 0x54, 0xb2, 0x9c, 0xc5, 0xb8, 0x5a, 0x5c,
+    0xee, 0x6b, 0xae, 0x0c, 0xf1, 0xc2, 0x1f, 0x27,
+    0x31, 0xec, 0xe2, 0x01, 0x3d, 0xc3, 0xfb, 0x7c,
+    0x8d, 0x21, 0x65, 0x4b, 0xb1, 0x61, 0xb4, 0x63,
+    0x96, 0x2c, 0xa1, 0x9e, 0x8c, 0x65, 0x4f, 0xf2,
+    0x4c, 0x94, 0xdd, 0x28, 0x98, 0xde, 0x12, 0x05,
+    0x1f, 0x1e, 0xd0, 0x69, 0x22, 0x37, 0xfb, 0x02,
+    0xb2, 0xf8, 0xd1, 0xdc, 0x1c, 0x73, 0xe9, 0xb3,
+    0x66, 0xb5, 0x29, 0xeb, 0x43, 0x6e, 0x98, 0xa9,
+    0x96, 0xee, 0x52, 0x2a, 0xef, 0x86, 0x3d, 0xd5,
+    0x73, 0x9d, 0x2f, 0x29, 0xb0
+};
+
+/*
+ * @brief generate a key pair from an initial string and check public
+ * @param kem_id the KEM to use (RFC9180 code point)
+ * @ikm is the initial key material buffer
+ * @ikmlen is the length of ikm
+ * @pub is the public key buffer
+ * @publen is the length of the public key
+ * @return 1 for good, other otherwise
+ *
+ * This calls OSSL_HPKE_keygen specifying only the IKM, then
+ * compares the key pair values with the already-known values
+ * that were input.
+ */
+static int test_hpke_one_ikm_gen(uint16_t kem_id,
+                                 unsigned char *ikm, size_t ikmlen,
+                                 unsigned char *pub, size_t publen)
+{
+    int hpke_mode = OSSL_HPKE_MODE_BASE;
+    OSSL_HPKE_SUITE hpke_suite = OSSL_HPKE_SUITE_DEFAULT;
+    unsigned char lpub[OSSL_HPKE_MAXSIZE];
+    size_t lpublen = OSSL_HPKE_MAXSIZE;
+    EVP_PKEY *sk = NULL;
+
+    hpke_suite.kem_id = kem_id;
+    if (OSSL_HPKE_keygen(testctx, NULL, hpke_mode, hpke_suite,
+                         ikm, ikmlen, lpub, &lpublen, &sk) != 1) {
+        return - __LINE__;
+    }
+    if (sk == NULL)
+        return - __LINE__;
+    EVP_PKEY_free(sk);
+    if (lpublen != publen)
+        return - __LINE__;
+    if (memcmp(pub, lpub, publen))
+        return - __LINE__;
+
+    return 1;
+}
+
+static int test_hpke_ikms(void)
```

Comment:
you should probably use ADD_ALL_TESTS(func, count) instead of ADD_TEST().
And make your functions just do one test... 
void test_hpke_ikms(int tstid) //where tstid ranges from 0..count-1
{
}